File: sdk/php/src/Resources/BareMetal.php
<?php

namespace HiTechCloud\SDK\Resources;

class BareMetal extends BaseResource
{
    /**
     * Get reinstall templates
     */
    public function getReinstallTemplates(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/reinstall/templates');
    }

    /**
     * Cancel diagnostics
     */
    public function cancelDiag(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/diag/cancel');
    }

    /**
     * Get diag templates
     */
    public function getDiagTemplates(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/diag/templates');
    }

    /**
     * Get diagnostics
     */
    public function getDiag(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/diag');
    }

    /**
     * Run diagnostics
     */
    public function runDiag(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/diag', $data);
    }

    /**
     * Get rescue templates
     */
    public function getRescueTemplates(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/rescue/templates');
    }

    /**
     * Get rescue mode
     */
    public function getRescue(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/rescue');
    }

    /**
     * Start rescue mode
     */
    public function startRescue(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/rescue', $data);
    }

    /**
     * Cancel rescue mode
     */
    public function cancelRescue(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/rescue/cancel');
    }

    /**
     * Get server info
     */
    public function getInfo(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/info');
    }

    /**
     * Set hostname
     */
    public function setHostname(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/hostname', $data);
    }

    /**
     * List IPs
     */
    public function listIps(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/ips');
    }

    /**
     * Add IP
     */
    public function addIp(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/ips', $data);
    }

    /**
     * List VLANs
     */
    public function listVlans(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/vlans');
    }

    /**
     * Get IP details
     */
    public function getIp(int $serviceId, string $ip): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/ips/' . $ip);
    }

    /**
     * Update IP
     */
    public function updateIp(int $serviceId, string $ip, array $data): array
    {
        return $this->http->put('/api/service/' . $serviceId . '/ips/' . $ip, $data);
    }

    /**
     * Get reinstall info
     */
    public function getReinstall(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/reinstall');
    }

    /**
     * Reinstall server
     */
    public function reinstall(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/reinstall', $data);
    }

    /**
     * Get server status
     */
    public function getStatus(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/status');
    }

    /**
     * Power action
     */
    public function powerAction(int $serviceId, string $action): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/power/' . $action);
    }

    /**
     * Get server stock
     */
    public function getServerStock(array $params = []): array
    {
        return $this->http->get('/api/serverstock', $params);
    }

}

File: sdk/csharp/Resources/BareMetal.cs
namespace HiTechCloud.SDK.Resources;

public class BareMetal : BaseResource
{
    public BareMetal(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ReinstallTemplatesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/reinstall/templates", id));
    }

    public async Task<Dictionary<string, object?>> CancelDiagAsync(int id)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/diag/cancel", id), null);
    }

    public async Task<Dictionary<string, object?>> DiagTemplatesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/diag/templates", id));
    }

    public async Task<Dictionary<string, object?>> DiagAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/diag", id));
    }

    public async Task<Dictionary<string, object?>> RunDiagAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/diag", id), data);
    }

    public async Task<Dictionary<string, object?>> RescueTemplatesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/rescue/templates", id));
    }

    public async Task<Dictionary<string, object?>> RescueAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/rescue", id));
    }

    public async Task<Dictionary<string, object?>> EnterRescueAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/rescue", id), data);
    }

    public async Task<Dictionary<string, object?>> CancelRescueAsync(int id)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/rescue/cancel", id), null);
    }

    public async Task<Dictionary<string, object?>> InfoAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/info", id));
    }

    public async Task<Dictionary<string, object?>> SetHostnameAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/hostname", id), data);
    }

    public async Task<Dictionary<string, object?>> IpsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/ips", id));
    }

    public async Task<Dictionary<string, object?>> AddIpAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/ips", id), data);
    }

    public async Task<Dictionary<string, object?>> VlansAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vlans", id));
    }

    public async Task<Dictionary<string, object?>> IpAsync(int id, string ip)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/ips/{ip}", id, ip));
    }

    public async Task<Dictionary<string, object?>> UpdateIpAsync(int id, string ip, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{id}/ips/{ip}", id, ip), data);
    }

    public async Task<Dictionary<string, object?>> ReinstallAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/reinstall", id));
    }

    public async Task<Dictionary<string, object?>> DoReinstallAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/reinstall", id), data);
    }

    public async Task<Dictionary<string, object?>> StatusAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/status", id));
    }

    public async Task<Dictionary<string, object?>> PowerAsync(int id, string action, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/power/{action}", id, action), data);
    }

    public async Task<Dictionary<string, object?>> ServerStockAsync()
    {
        return await Http.GetAsync("/api/serverstock");
    }
}
